includes/class-wc-payments-db.php

```diff
@@ -38,28 +38,26 @@ public function order_from_charge_id( $charge_id ) {
 	 * @return array[]
 	 */
 	public function orders_with_charge_id_from_charge_ids( array $charge_ids ): array {
-		global $wpdb;
-
-		$charge_id_placeholder = implode( ',', array_fill( 0, count( $charge_ids ), '%s' ) );
 
 		// The order ID is saved to DB in `WC_Payment_Gateway_WCPay::process_payment()`.
-		$results = $wpdb->get_results(
-			$wpdb->prepare(
-				// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared,WordPress.DB.PreparedSQLPlaceholders.UnfinishedPrepare
-				"SELECT DISTINCT ID as order_id, meta.meta_value as charge_id FROM $wpdb->posts as posts LEFT JOIN $wpdb->postmeta as meta ON posts.ID = meta.post_id WHERE meta.meta_key = '_charge_id' AND meta.meta_value IN ($charge_id_placeholder)",
-				$charge_ids
-			)
+		$orders = wc_get_orders(
+			[
+				'meta_key'     => self::META_KEY_CHARGE_ID, //phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_key
+				'meta_value'   => $charge_ids, //phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_value
+				'meta_compare' => 'IN',
+			]
 		);
 
 		return array_map(
-			function ( stdClass $row ) : array {
+			function ( WC_Order $order ) : array {
 				return [
-					'order'     => $this->order_from_order_id( $row->order_id ),
-					'charge_id' => $row->charge_id,
+					'order'     => $this->order_from_order_id( (string) $order->get_id() ),
```

Comment:
Shouldn't we be getting the `order` itself from `wc_get_orders` in the first place? Why do we need to call this by extracting the id from it?

Reply:
Sorry, missed that. Fixed now.